Add Python 3.7 Support

### Current behavior

- What is the problem? 

Project doesn't install and run on Python 3.7

### Expected behavior

- What should be happening?

Project should install and run on Python 3.7

When running travis on neo-python+Python 3.7, I get this failure: 

FAIL: test_build_contract_and_destroy (SmartContract.tests.test_migrate_destroy.ContractMigrateTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/c/users/jseag/Neo/neo-python-37/neo/SmartContract/tests/test_migrate_destroy.py", line 71, in test_build_contract_and_destroy
    self.assertEqual(len(results), 1)
AssertionError: 7 != 1

This is the only error. I am continuing to work out the differences in functionality and specifically why this error would occur. This is my Python 3.7 branch: https://github.com/jseagrave21/neo-python/tree/patch-3?files=1

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -4,7 +4,9 @@ Changelog
 All notable changes to this project are documented in this file.
 
 [0.7.8-dev] 2018-xx-xx
+----------------------
 - Prefix ``vin`` JSON output format to match C#
+- Update ``neo-boa`` to v0.5.0 for Python 3.7 compatibility
 
 
 [0.7.7] 2018-08-23

--- a/requirements.txt
+++ b/requirements.txt
@@ -34,7 +34,7 @@ memory-profiler==0.52.0
 mmh3==2.5.1
 mock==2.0.0
 mpmath==1.0.0
-neo-boa==0.4.9
+neo-boa==0.5.0
 neo-python-rpc==0.2.1
 neocore==0.5.1
 pbr==4.1.1

